src/engine.c

```diff
@@ -381,12 +388,15 @@ ENGINE_pset(ENGINE* engine, int64_t x, int64_t y, uint32_t c) {
 
       uint16_t newA = (0xFF000000 & c) >> 24;
 
-      uint16_t oldR = (255-newA) * ((0x000000FF & current));
-      uint16_t oldG = (255-newA) * ((0x0000FF00 & current) >> 8);
-      uint16_t oldB = (255-newA) * ((0x00FF0000 & current) >> 16);
-      uint16_t newR = newA * ((0x000000FF & c));
-      uint16_t newG = newA * ((0x0000FF00 & c) >> 8);
-      uint16_t newB = newA * ((0x00FF0000 & c) >> 16);
+      int oldR, oldG, oldB, newR, newG, newB;
```

Comment:
This should probably be uint8_t, since DOME prefers to use fixed size types where possible.

Reply:
In the beginning all of those types (here and in the other place) were `uint8_t`. But then I was an overflow problem with the calculation of the alpha. I thought about multiplying by a fraction or even casts, but the decided that the best will be to just fall back to `int`.

Reply:
What circumstances result in an overflow problem? I'd like to resolve those in a separate PR.

Reply:
You multiply by the alpha (then divide by 255). As I said, you can instead multiply by `1.0 / newA`.

Reply:
I would need to profile if the use of a floating point operation has a significant impact, but that's something for a different PR.

For now, please at least use `uint32_t` rather than generic `int`, in keeping with the rest of the engine's code style.

Reply:
Changed to `uint16_t`, as is consistent with `newA`.